ZuluContent/Items/Skill Items/Fishing/Misc/SpecialFishingNet.cs

```diff
@@ -292,146 +256,286 @@ protected void Spawn(Point3D p, Map map, BaseCreature spawn)
+                            from.AddToBackpack(sosBottle);
+                        }
+                        else
+                        {
+                            var pouch = new Pouch();
+                            pouch.AddItem(new Gold(Utility.Random(100) + 250));
+                            from.AddToBackpack(pouch);
+                            from.SendSuccessMessage("You find an old backpack!");
+                        }
+                    }
+                    break;
+                }
+                case 3:
+                {
+                    var shell = GetRandomShell(from);
+                    from.AddToBackpack(shell);
+                    from.SendSuccessMessage("You find something valuable!");
+                    break;
+                }
+                case 4:
+                case 5:
+                {
+                    var magicFish = GetFish(from);
+                    if (magicFish == null)
+                    {
+                        from.AddToBackpack(new Fish(Utility.RandomMinMax(6, 12)));
+                        from.ShilCheckSkill(SkillName.Fishing, points: 100);
+                    }
+                    else
+                    {
+                        magicFish.Amount = Utility.RandomMinMax(1, 2);
+                        from.AddToBackpack(magicFish);
+                        from.ShilCheckSkill(SkillName.Fishing, points: 150);
+                    }
+                    from.SendSuccessMessage("You caught some fish.");
+                    break;
+                }
+                case 6:
+                {
+                    if (Utility.Random(10) > 8)
+                    {
+                        from.AddToBackpack(new FishingTrophy1(from));
+                        from.SendSuccessMessage("You catch a trophy!");
+                    }
+                    else if (Utility.Random(10) > 8)
+                    {
+                        from.AddToBackpack(new FishingTrophy2(from));
+                        from.SendSuccessMessage("You catch a trophy!");
+                    }
+                    else
+                    {
+                        var bag = new Bag();
+                        bag.AddItem(new Gold(Utility.Random(200) + 150));
+                        from.AddToBackpack(bag);
+                        from.SendSuccessMessage("You find an old bag!");
+                    }
 
-                spawn.Combatant = from;
+                    break;
+                }
+                case 7:
+                case 8:
+                {
+                    var remains = GetRemains(from);
+                    from.AddToBackpack(remains);
+                    from.SendSuccessMessage("You find human skeleton remains!");
+                    break;
+                }
+                case 9:
+                {
+                    var seaSerpent = new SeaSerpent();
+                    Spawn(from.Location, from.Map, seaSerpent);
```

Comment:
This should probably be the water tile they selected? Otherwise you'll have sea serpents on land!

Same with the other spawn above.

Reply:
I decided to just use the code from before which does some randomization and makes sure the tiles are sea tiles (check the Spawn function out). The fishing code above only spawns a waterelemental or walrus so all good